CONTRIBUTING.md

```diff
@@ -0,0 +1,529 @@
+classroom, while the reference implementation is like a textbook.  While a more
+adventurous student may skip class to rely on reading the book, every class
+needs an accompanying textbook for supporting material.  The accompanying
+reference implementation can be as simple as reproducing the code cells in a
+single script, or potentially more complex if supplemented by helper
+functions/classes/modules.
+
+### Value Comparison
+
+Each reference implementation and interactive tutorial should conclude with a
+testable value comparison, which makes use of Psi4's `compare_values()`
+function (documented
+[here](http://psicode.org/psi4manual/1.1/api/psi4.driver.compare_values.html?highlight=compare_values#psi4.driver.compare_values)):
+
+```python
+psi4.driver.compare_values(expected, computed, digits, label, exitonfail=True)
+```
+
+where `expected` should be call to Psi4 to compute the desired value (if Psi4
+has such capability) or a hard-coded value (if Psi4 does not have such
+capabilities), and `computed` should be the final output of the reference
+implementation or tutorial.  Value comparison statements will serve an
+important role within [continuous
+integration](#testing-and-continuous-integration), as thse statements are the
+primary means by which the successful execution of a script or tutorial is
+determined.  
+
+## Styleguides
+
+### Git Commit Messages
+
+In addition to the conventions for writing good git commit messages
+[here](https://chris.beams.io/posts/git-commit/), commits to Psi4NumPy should
+add the following codes to the beginning of the subject line to categorize the
+nature of the commit:
+
+| Code | Category Description |
+-------|-----------------------
+| RIMP | New reference implementation |
+| TUT | New interactive tutorial |
+| DEV | New developer tool or utility |
+| ENH | Enhancement of any existing implementations or tutorials |
+| UPD | Update to new Psi4/NumPy/SciPy compatibility |
+| MAINT | Maintenance commit (typos, repo reorganization, etc.) |
+| STY | Style convention changes (PEP8, headers, etc.) |
+| TST | Related to tests of the codebase |
+| REF | Adding to/modifying any references/citations in tutorials, scripts, or READMEs |
+| DOC | Documentation (README additions, etc. other than REF) |
+| REL | Related to the release of Psi4NumPy |
+
+### Python Styleguide
+
+All Python code should should conform to the
+[PEP8](https://www.python.org/dev/peps/pep-0008/) Python style guide
+conventions.  This code should be compliant with Python versions 2.7/3.5/3.6,
+meaning that `print` must be called as a function, i.e., `print('Compliant with
+Python 3')` and not `print 'This will break Python 3!'`.
+
+### Citation Styleguide
+
+For both reference implementations and interactive tutorials, a complete list
+of citations from which the content was drawn must be provided.  Such citation
+lists should contain entries of the form (in Markdown):
+```
```

Comment:
Some of this may need a TLDR section, which for this could be: "Citations, including equation numbers, should be provided in close proximity (docstrings or comments) to the code. It can also be handy to collect all citations into the nearest README.md (for reference implementations) or final cell (for tutorials).